core/src/main/java/org/apache/gravitino/GravitinoEnv.java

```diff
@@ -348,28 +352,30 @@ private void initGravitinoServerComponents() {
 
     // Create and initialize metalake related modules
     MetalakeDispatcher metalakeManager = new MetalakeManager(entityStore, idGenerator);
+    MetalakeHookDispatcher metalakeHookDispatcher = new MetalakeHookDispatcher(metalakeManager);
     MetalakeNormalizeDispatcher metalakeNormalizeDispatcher =
-        new MetalakeNormalizeDispatcher(installDispatcherHooks(metalakeManager));
+        new MetalakeNormalizeDispatcher(metalakeHookDispatcher);
     this.metalakeDispatcher = new MetalakeEventDispatcher(eventBus, metalakeNormalizeDispatcher);
 
     // Create and initialize Catalog related modules
     this.catalogManager = new CatalogManager(config, entityStore, idGenerator);
+    CatalogHookDispatcher catalogHookDispatcher = new CatalogHookDispatcher(catalogManager);
     CatalogNormalizeDispatcher catalogNormalizeDispatcher =
-        new CatalogNormalizeDispatcher(installDispatcherHooks((CatalogDispatcher) catalogManager));
+        new CatalogNormalizeDispatcher(catalogHookDispatcher);
```

Comment:
It seems that we have wrapped the dispatcher too many times.

Reply:
Known issues. But we don't have better solution.

Reply:
Maybe we can put normalizeDispatcher to HookDispatcher in the future.

Reply:
I also feel the current `Dispatcher` mechanism inelegant.
I create a issue track it 
- https://github.com/apache/gravitino/issues/4446